src/utils/utils.ts

```diff
@@ -48,10 +56,100 @@ export function createElement<K extends keyof HTMLElementTagNameMap>(
 /**
  * Similar to Object.assign, but skips properties that already exist in `obj`.
  */
-export function assignIfNotExists<T, E>(obj: T, other: E): T & Omit<E, keyof T> {
+export function assignIfNotExists<T, E>(
+    obj: T, other: E
+): T & Omit<E, keyof T> {
     const o = obj as any;
     for (const [key, val] of Object.entries(other)) {
         if (!(key in obj)) o[key] = val;
     }
     return o;
 }
+
+
+// This function was taken from the now deprecated dagrejs library, see:
```

Comment:
we can't do this without a proper license / attribution

Reply:
How / where would we need to attribute?